src/ensembl/src/content/app/browser/track-panel/trackPanelActions.ts

```diff
@@ -99,10 +106,62 @@ export const changeTrackPanelModalViewForGenome: ActionCreator<
   );
 };
 
+export const updatePreviouslyViewedObjectsAndSave: ActionCreator<
+  ThunkAction<void, any, null, Action<string>>
+> = () => (dispatch, getState: () => RootState) => {
```

Comment:
When is this action called — **after** we have switched to the new focus object in redux store or **before** we have? From the code, it seems that the action is called after redux store has been updated to the new focus object, but shouldn't it be called before, in order to capture the track states of the previous object?

I have also noticed that you call `updateTrackStates` in individual components; but if this action were called right before the focus object has changed, it could have incorporated snapshotting of track states within the code of this action, and there would be no need to call `updateTrackStates` from individual components? At least, I hope there wouldn't.